### PR TITLE
kdenlive

### LINKED ISSUE #16837: [Package]: kdenlive
### Why is it worth to add this package?

I want to use a video editor inside termux. In android, the most video editor needs a premium subscription. But kdenlive is free software. I think it could work in termux. I know that it is available only in ubuntu and arch, so I understand why is not in termux repositories.

### Home page URL

https://kdenlive.org

### Source code URL

https://github.com/KDE/kdenlive

### Packaging policy acknowledgement

- [X] I certify that I have read [Termux Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy) and understand that my request will be denied in case of violation.

### Additional information

This is the installation guide:
https://github.com/KDE/kdenlive/blob/master/dev-docs/build.md

### PR DESCRIPTION
**Pull request by knyipab**
Should bring us closer to a complete KDE. Also prove that `qt6-qt5compat` #21573 is working. 

~Some dirty hacks applied on `kf6-kconfig` to build kdenlive. Help wanted to address such issue, see #21647.~ Done

Closes #16837

### CODE REVIEW

**Comment by twaik** [comment]
I do not think using separate package for building with cross-compiler is a bad idea. There are a lot of packages like this. You can find a lot of `*-cross*.subpackage.sh` buildscripts in termux packages. But probably it will be better to move `x11-packages/kf6-kconfig-cross/build.sh` content to the regular `kf6-config` build script with moving cross-building stuff to subpackage. This way we will be sure both regular and cross-subpackage have the same version and updated simultaneously.

**Comment by Biswa96** [comment]
Before merging this pull request, we need to figure out how to manage auto-update with a group of packages instead of individual packages. Otherwise, we would get multiple kf6 auto-update issues. For example, we recently get qt6 auto-update issues which would not happen if those packages are updated in a specific order.

I can not figure out how to enable auto-update for groups. If anyone will implement that I can try to add the package groups e.g. qt5/6, kf5/6, gstreamer etc.

**Comment by twaik** [comment]
If the only thing you need in kf6-kconfig-cross-tools is `kconfig_version.h` you may just generate it, it is pretty simple.
```
// This file was generated by ecm_setup_version(): DO NOT EDIT!

#ifndef KCONFIG_VERSION_H
#define KCONFIG_VERSION_H

#define KCONFIG_VERSION_STRING "5.115.0"
#define KCONFIG_VERSION_MAJOR 5
#define KCONFIG_VERSION_MINOR 115
#define KCONFIG_VERSION_PATCH 0
#define KCONFIG_VERSION ((5<<16)|(115<<8)|(0))

#endif
```
So it will be a part of `kf6-kconfig` subpackage.

**Review by twaik — APPROVED** [review]

**Comment by twaik** [comment]
Closes #16837

**Comment by hansm629** [comment]
@knyipab 
I'm eagerly waiting for the `Kdenlive` merge to be completed~

This is unrelated to the current pull request, but in the future, do you think there's a possibility that programs like `Pencil2D`, `Kolourpaint`, and `Okular` will also be ported to Termux?

**Comment by Biswa96** [comment]
There are some git conflict with recent kf6 changes. Here are two things to follow.
1. I have added the following two cmake options for all kf6 package. Please add those here too.

```
-DKDE_INSTALL_QMLDIR=lib/qt6/qml
-DKDE_INSTALL_QTPLUGINDIR=lib/qt6/plugins
```

2. I have added the version constrain `(>= ${TERMUX_PKG_VERSION})` for **only** kf6 dependencies in TERMUX_PKG_DEPENDS and TERMUX_PKG_BUILD_DEPENDS. Please add that also. This would probably help with multiple auto update issues.

**Comment by twaik** [comment]
image
Probably we should add ffplay and mediainfo to dependencies.
What about missing modules?

**Comment by twaik** [comment]
I succeeded to import video and play it. Can you please fix these warnings before we merge it?

**Comment by twaik** [comment]
Since fine by now to me.
It will be merged in 72 hours if no one minds or when it gets 2+ other approves.

**Review by twaik — APPROVED** [review]

File: packages/libvoikko/build.sh
Comment (TomJo2000) [inline packages/libvoikko/build.sh:1]: Since this is an indirect dependency it would be good to mention what package requires it.
```suggestion
# Build time dependency for kf6-sonnet
TERMUX_PKG_HOMEPAGE="http://voikko.sourceforge.net"
```

File: packages/convertlit/build.sh
Comment (TomJo2000) [inline packages/convertlit/build.sh:1]: Since this is an indirect dependency it would be good to mention what package requires it.
```suggestion
# Dependency for ebook-tools
TERMUX_PKG_HOMEPAGE='http://www.convertlit.com/'
```

File: packages/ebook-tools/build.sh
Comment (TomJo2000) [inline packages/ebook-tools/build.sh:1]: Since this is an indirect dependency it would be good to mention what package requires it.
```suggestion
# Dependency for kf6-kfilemetadata
TERMUX_PKG_HOMEPAGE='https://sourceforge.net/projects/ebook-tools/'
```

**Review by TomJo2000 — APPROVED** [review]
LGTM.

It'd be appreciated if you could set yourself as the maintainer for at least some of the core packages for this PR.
e.g. `kdenlive` and `mlt`.

**Comment by truboxl** [comment]
Can you stop merging? There are plenty of changes I am requesting. Now I have to do it manually.

**Comment by twaik** [comment]
You did not post reviews in the last 72 hours so I thought it will be fine to merge it. Sorry.

**Review by truboxl — COMMENTED** [review]
Ugh... Here are all the changes I am requesting.

File: x11-packages/kf6-kirigami/build.sh
Comment (twaik) [reply x11-packages/kf6-kirigami/build.sh:8]: `qt6-qtsvg` not needed here, right?

**Comment by twaik** [comment]
It is my fault I merged without taking a closer look, I'll fix some of these.

File: packages/convertlit/build.sh
Comment (twaik) [reply packages/convertlit/build.sh:9]: Fixed by db946542e20a14fd0ad19fdc3eae918a834bd2c4.

File: x11-packages/kf6-breeze-icons/build.sh
Comment (twaik) [reply x11-packages/kf6-breeze-icons/build.sh:1]: Done in https://github.com/termux/termux-packages/commit/085ae81991cbe0f81e1c9a09f9c8280492b7311e.

File: x11-packages/kf6-breeze-icons/build.sh
Comment (twaik) [reply x11-packages/kf6-breeze-icons/build.sh:42]: Done.

File: x11-packages/kf6-breeze-icons/build.sh
Comment (twaik) [reply x11-packages/kf6-breeze-icons/build.sh:56]: Done.

**Comment by twaik** [comment]
Resolved everything except kdenlive. Turns out it does not want to be built separately from kf6-kconfig, fixing it.

**Comment by hansm629** [comment]
@knyipab 
Hello Sir,
Thank you so much for developing `Kdenlive` for Termux!

I have one question. `Shotcut`, for example, supports `multiple languages`, allowing for language changes (ex : Korean).

Kdenlive has a language setting menu, but only English is available as a selectable language.
Is this something that could possibly be improved?

**Comment by Biswa96** [comment]
> Is this something that could possibly be improved?

It is possible to add the translations. The build process needs kdoctools. There could be some path related quirks but it's doable.

**Comment by hansm629** [comment]
@Biswa96 
Could you possibly add it in the next update version?

**Comment by Biswa96** [comment]
> Kdenlive has a language setting menu, but only English is available as a selectable language.

It seems that locales are removed from packages explicitly https://github.com/termux/termux-packages/blob/bbb879d44170ccbf07bdd65ac065a29f32aae243/scripts/build/termux_step_massage.sh#L30. I do not know the reason for that exactly (probably to reduce package size).

**Comment by hansm629** [comment]
@knyipab @Biswa96 
Could you please add locales in the next version?

**Pull request merged**
